python/lsst/ts/mtmount/mtmount_csc.py

```diff
@@ -1194,23 +1263,25 @@ async def _basic_send_command(self, command, *, timeout=None):
         self.command_futures_dict[command.sequence_id] = command_futures
         command_bytes = command.encode()
         self.log.log(LOG_LEVEL_COMMANDS, "Send command %s", command_bytes)
+        self.command_history.append(command_bytes)
         await self.client.write(command_bytes)
         initial_timeout = self.config.ack_timeout if timeout is None else timeout
         new_timeout = await asyncio.wait_for(command_futures.ack, initial_timeout)
+        timeout_buffer = TIMEOUT_BUFFER if timeout is None else TIMEOUT_BUFFER + timeout
         if not command_futures.done.done():
             try:
                 await asyncio.wait_for(
                     command_futures.done,
                     timeout=(
-                        (TIMEOUT_BUFFER + new_timeout)
+                        (timeout_buffer + new_timeout)
                         if new_timeout is not None
                         else None
                     ),
                 )
             except asyncio.TimeoutError:
                 self.command_futures_dict.pop(command.sequence_id, None)
                 raise asyncio.TimeoutError(
-                    f"Command {command} timed out after {new_timeout + TIMEOUT_BUFFER} seconds"
+                    f"Command {command} timed out after {new_timeout + timeout_buffer} seconds."
```

Comment:
This doesn't take the case where timeout==None into account.

Reply:
But the timeout is calculated again in lines 1275-1279 and there it can be `None` and that case is not covered by line 1284.

Reply:
interesting, you are right.. I already merged it tough... :( will have to fix it later...